### PR TITLE
add (Base)ExceptionGroup

### LINKED ISSUE #6632: Add new builtin exception types from Python 3.11 - BaseExceptionGroup & ExceptionGroup
Introduced in [Python 3.11][1] by [PEP 654][2]

```python-repl
>>> BaseExceptionGroup
<class 'BaseExceptionGroup'>

>>> ExceptionGroup
<class 'ExceptionGroup'>
```

[1]: https://docs.python.org/3.11/whatsnew/3.11.html
[2]: https://www.python.org/dev/peps/pep-0654/

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Fixes #6632

### CODE REVIEW

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -1651,3 +1651,19 @@ class ResourceWarning(Warning): ...
 
 if sys.version_info >= (3, 10):
     class EncodingWarning(Warning): ...
+
+if sys.version_info >= (3, 11):
+    _SplitCondition = type[BaseException] | tuple[type[BaseException, ...]] | Callable[[BaseException], bool]
+    class BaseExceptionGroup(BaseException):
+        def __new__(cls, __message: str, exceptions: Sequence[BaseException]) -> BaseExceptionGroup | ExceptionGroup: ...
```
Comment (srittau) [inline stdlib/builtins.pyi:1658]: According to the PEP, `exceptions` is also positional-only. Is the implementation different?

**Review by srittau — APPROVED** [review]
One question below, but LGTM.

File: stdlib/builtins.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stdlib/builtins.pyi:1658]: Oops, my mistake.
```suggestion
        def __new__(cls, __message: str, __exceptions: Sequence[BaseException]) -> BaseExceptionGroup | ExceptionGroup: ...
```

**Comment by agronholm** [comment]
A couple questions:

1. Why is the second argument to `ExceptionGroup` named differently than that of `BaseExceptionGroup`? This definition makes it look like it could be passed as a keyword argument, which it can't be.
2. Why is the return type of the `exceptions` property typed as a `Sequence` when it always returns a tuple, regardless of the type of the initial exceptions sequence?

I'm happy to submit a follow-up PR to fix these.

**Comment by JelleZijlstra** [comment]
@agronholm good catch about the `exceptions` argument. They should both be positional-only.

I think I used `Sequence` because the PEP (https://www.python.org/dev/peps/pep-0654/#exceptiongroup-and-baseexceptiongroup) says it's a sequence, but you're right that we prefer concrete types like tuple in typeshed.

**Comment by agronholm** [comment]
The PEP doesn't really define the `exceptions` property but it is mentioned there. The constructor argument `__exceptions` is correct here, as it can be any sequence, but the implementation turns it into a tuple in the getter of `exceptions`.

**Comment by JelleZijlstra** [comment]
Yes, the summary is that you're right on both counts :) Feel free to submit a PR!

**Comment by agronholm** [comment]
Done: https://github.com/python/typeshed/pull/6895

**Comment by agronholm** [comment]
I noticed another issue: `BaseExceptionGroup` and `ExceptionGroup` are actually generic, but this is not reflected in the stubs. Thankfully `tuple` is covariant so there is no conflict in the `exceptions` property. I'll send another PR.

**Pull request merged**